olive/evaluator/olive_evaluator.py

```diff
@@ -148,48 +165,68 @@ def get_user_config(metric: Metric):
+                sub_type_metric_value = AUC(metric_config).measure(preds, targets)
+            else:
+                raise TypeError(f"{sub_type} is not a accuracy metric supported")
+            metric_res[sub_type.name] = SubTypeMetricResult(
+                value=sub_type_metric_value,
+                priority=sub_type.priority,
+                higher_is_better=sub_type.higher_is_better,
+            )
+        return MetricResult.parse_obj(metric_res)
 
     @staticmethod
-    def compute_latency(metric: Metric, latencies: Any) -> float:
+    def compute_latency(metric: Metric, latencies: Any) -> MetricResult:
         """
         Compute latency metrics
         """
-        if metric.sub_type == LatencySubType.AVG:
-            return round(sum(latencies) / len(latencies) * 1000, 5)
-        elif metric.sub_type == LatencySubType.MAX:
-            return round(max(latencies) * 1000, 5)
-        elif metric.sub_type == LatencySubType.MIN:
-            return round(min(latencies) * 1000, 5)
-        elif metric.sub_type == LatencySubType.P50:
-            return round(np.percentile(latencies, 50) * 1000, 5)
-        elif metric.sub_type == LatencySubType.P75:
-            return round(np.percentile(latencies, 75) * 1000, 5)
-        elif metric.sub_type == LatencySubType.P90:
-            return round(np.percentile(latencies, 90) * 1000, 5)
-        elif metric.sub_type == LatencySubType.P95:
-            return round(np.percentile(latencies, 95) * 1000, 5)
-        elif metric.sub_type == LatencySubType.P99:
-            return round(np.percentile(latencies, 99) * 1000, 5)
-        elif metric.sub_type == LatencySubType.P999:
-            return round(np.percentile(latencies, 99.9) * 1000, 5)
-        else:
-            raise TypeError(f"{metric.sub_type} is not a supported latency metric")
+        latency_metrics = {
+            LatencySubType.AVG: round(sum(latencies) / len(latencies) * 1000, 5),
+            LatencySubType.MAX: round(max(latencies) * 1000, 5),
+            LatencySubType.MIN: round(min(latencies) * 1000, 5),
+            LatencySubType.P50: round(np.percentile(latencies, 50) * 1000, 5),
+            LatencySubType.P75: round(np.percentile(latencies, 75) * 1000, 5),
+            LatencySubType.P90: round(np.percentile(latencies, 90) * 1000, 5),
+            LatencySubType.P95: round(np.percentile(latencies, 95) * 1000, 5),
+            LatencySubType.P99: round(np.percentile(latencies, 99) * 1000, 5),
+            LatencySubType.P999: round(np.percentile(latencies, 99.9) * 1000, 5),
+        }
+        metric_res = {}
+        for sub_type in metric.sub_types:
+            metric_res[sub_type.name] = SubTypeMetricResult(
+                value=latency_metrics[sub_type.name],
+                priority=sub_type.priority,
+                higher_is_better=sub_type.higher_is_better,
+            )
+        return MetricResult.parse_obj(metric_res)
+
+    def _get_latency_config_from_metric(self, metric: Metric):
```

Comment:
make it method of Metric so all other place could reuse it?

Reply:
Agree.~